PR: https://github.com/odoo/odoo/pull/253079

From: 3259a7fae4ca4ee98f7e02da5109c43cd9e77f69
From: Tatiana Skvortsova (taskv)
Date: 2026-04-13 06:36:16

Breaking data model changes scores: del:4 + add:0, change matches:
-    departure_do_unassign_equipment = fields.Boolean(related='version_id.departure_do_unassign_equipment',
-    do_unassign_equipment = fields.Boolean("Free Equiments", default=True, help="Unassign Employee from Equipments")
-    _inherit = 'hr.version'
-    departure_do_unassign_equipment = fields.Boolean(related='departure_id.do_unassign_equipment')

Total Changes: 23

[IMP] l10n_be_hr_payroll: Improve End of Collaboration UX

Improve the UX of the "End of Collaboration" procedure, particularly for Belgium.

The option to select which actions are executed upon employee departure
(delete user, archive employee, unassign car, etc.) has been removed.

All actions are now executed automatically. However, it is still possible
to define the date on which they will be applied.

Related PRs:
Enterprise: odoo/enterprise#110147
Upgrade: odoo/upgrade#9653

task-5958676

test

closes odoo/odoo#253079

Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_maintenance/models/__init__.py
+++ b/addons/hr_maintenance/models/__init__.py
@@ -3,5 +3,4 @@
 from . import equipment
 from . import hr_employee
 from . import hr_employee_departure
-from . import hr_version
 from . import hr_employee_public

--- a/addons/hr_maintenance/models/hr_employee.py
+++ b/addons/hr_maintenance/models/hr_employee.py
@@ -6,8 +6,6 @@ class HrEmployee(models.Model):
 
     equipment_ids = fields.One2many('maintenance.equipment', 'employee_id', groups="hr.group_hr_user")
     equipment_count = fields.Integer('Equipment Count', compute='_compute_equipment_count')
-    departure_do_unassign_equipment = fields.Boolean(related='version_id.departure_do_unassign_equipment',
-        inherited=True, readonly=False, groups="hr.group_hr_user")
 
     @api.depends('equipment_ids')
     def _compute_equipment_count(self):

--- a/addons/hr_maintenance/models/hr_employee_departure.py
+++ b/addons/hr_maintenance/models/hr_employee_departure.py
@@ -1,20 +1,15 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import models
 
 
 class HrEmployeeDeparture(models.Model):
     _inherit = 'hr.employee.departure'
 
-    do_unassign_equipment = fields.Boolean("Free Equiments", default=True, help="Unassign Employee from Equipments")
-
     def action_register(self):
         super().action_register()
-        departure_with_unassign = self.filtered(lambda dep: dep.do_unassign_equipment)
-        if not departure_with_unassign:
-            return
-        all_equipments = departure_with_unassign.employee_id.equipment_ids
-        for departure in departure_with_unassign:
+        all_equipments = self.employee_id.equipment_ids
+        for departure in self:
             equipments = departure.employee_id.equipment_ids
             if not equipments:
                 continue

--- a/addons/hr_maintenance/models/hr_version.py
+++ b/None
@@ -1,9 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class HrVersion(models.Model):
-    _inherit = 'hr.version'
-
-    departure_do_unassign_equipment = fields.Boolean(related='departure_id.do_unassign_equipment')
